update upgrade in place (#373)
filecoin-project/filecoin-docs: docs/how-to/join-a-network.md

## Upgrade in place
Here’s how to upgrade in place, with minimum impact:
* `Back up your wallet`: just in case
  ```bash
  lotus wallet export t3… > path/wallet.keyinfo
  ```
* `Find a good upgrade window`: If you are running miner(s), check if your miner is safe to shut down and restart: 
  ```bash
  lotus-miner proving info
  ```
  If any deadline shows a block height in the past, do not restart. See examples below.

>     In the following example, `Deadline Sectors` is 0 which means no sectors are currently being challenged in this deadline. This deadline closes in 7 minutes, as indicated by `Deadline Close`. This means the miner can be safely restarted in the next 7 minutes, before the next deadline reaches.

Reviewer: I don't think the bit about `7 minutes` is reliable here -- pretty sure Lotus is displaying a different value here. I'd take it out for the time being (see https://github.com/filecoin-project/lotus/issues/3630).

Author: updated

    Miner: t01000
    Current Epoch:           497
    Proving Period Boundary: 658
    Proving Period Start:    658 (in 1h20m30s)
    Next Period Start:       3538 (in 25h20m30s)
    Faults:      0 (0.00%)
    Recovering:  0
    Deadline Index:       0
    Deadline Sectors:     768
    Deadline Open:        658 (in 1h20m30s)
    Deadline Close:       718 (in 1h50m30s)
    Deadline Challenge:   638 (in 1h10m30s)
    Deadline FaultCutoff: 588 (in 45m30s)
    ```

    After you find a good window, move to the next step!
* `Stop your nodes(daemon, miner/workers)`:
  * run
    ```bash
    lotus daemon stop
    lotus-miner stop
    ```
* `Get the latest version and build the new binary`: 
  * Navigate  to your lotus directory
  * run:
    ```bash
    git pull && git checkout <tag>
    make all && make install
    ``` 
    This should preserve your config, wallets, and anything else on disk.
* `Start daemon`: 
  * run: 
    ```bash
    lotus daemon
    ``` 
  * Check your syncing status by running 
    ```bash
    lotus sync wait
    ``` 
    `Done` in the output indicates you have been synced up with the network!
* `Start your miner`: 
  ```bash
  lotus-miner run
  ```

> * `Start your worker`:

Reviewer: Nit: `Start your worker(s)`

* If you are running miner(s), check if your miner is safe to shut down and restart: `lotus-miner proving info`. If any deadline shows a block height in the past, do not restart. See examples below.
* Otherwise, you can safely restart your daemon and miner(s). This will replace and update the binary, and should preserve your config, wallets, and anything else on disk.

> In the following example, `Deadline Sectors` is 0 which means no sector is currently being challenged. And this dealine is going to closed in 7 minutues, as indicated by `Dealdline Close`. This means the miner can be safely restarted in the next 7 minutes, before the next deadline reaches.

Reviewer 2: ```suggestion
In the following example, `Deadline Sectors` is 0 which means no sectors are currently being challenged in this deadline. This deadline closes in 7 minutes, as indicated by `Deadline Close`. This means the miner can be safely restarted in the next 7 minutes, before the next deadline reaches.
```

Question for @travisperson @arajasek is it risky to restart with 7 minutes remaining?

Deadline FaultCutoff: 384 (58m0s ago)
```

> In this next example, this miner should **not** be shut down or restarted since there ar 768 sectors being challenged right now. 

Reviewer 2: ```suggestion
In this next example, this miner should **not** be shut down or restarted since there are 768 sectors being challenged right now. 
```

> In this next example, this miner should **not** be shut down or restarted since there are 768 sectors being challenged right now. 

Reviewer 3: 768 sectors will be challenged when the deadline opens in 1h20m30s.
